### PR TITLE
DB-11636: fix avg aggregator regression

### PR DESCRIPTION
**Pull request by jyuanca**
## Short Description
A regression due to serialization changes to avg aggregator

## Long Description
The SumAggregator included in a AvgAggregator can be one of LongBufferedSumAggregator, DecimalBufferedSumAggregator, or DoubleBufferedSumAggregator. When it is deserialized, we should find out its correct data type and construct the right aggregator.

## How to test
This is only reproducible if a grouped aggregation is executed a normal spark processor (not native spark). To disable native spark, set the following database property

```
call syscs_util.syscs_set_global_database_property('splice.execution.nativeSparkAggregationMode', 'off');
create table AG (a int, b real, c bigint, d decimal(15,2));
insert into AG values(1, 1, 2, 3);
insert into AG values(2, 1, 2, 3);
insert into AG values(3, 1, 2, 3);

insert into AG select a, b+1, c+1, d+1 from AG
insert into AG select a, b+1, c+1, d+1 from AG
insert into AG select a, b+1, c+1, d+1 from AG
insert into AG select a, b+1, c+1, d+1 from AG
```

This query should not fail with NPE
```
select a, avg(b), avg(c), avg(d) from ag --splice-properties useSpark=true
group by a order by a;
```

### CODE REVIEW

**Review by msirek — APPROVED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/DecimalBufferedSumAggregator.java [hunk 1]
```diff
@@ -61,6 +61,11 @@ public DecimalBufferedSumAggregator() { // SERDE
 
 	}
 
+	public DecimalBufferedSumAggregator(CatalogMessage.SystemAggregator agg) throws IOException, ClassNotFoundException {
+		this(64);
```
Comment (martinrupp) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/DecimalBufferedSumAggregator.java:65]: why 64?

**Review by martinrupp — APPROVED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/DecimalBufferedSumAggregator.java (hunk 1 above)
Comment (jyuanca) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/DecimalBufferedSumAggregator.java:65]: I'm copying the size from else where. Not sure why it was decided to be 64

**Review by carolp-503 — APPROVED** [review]
+1

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/DecimalBufferedSumAggregator.java (hunk 1 above)
Comment (carolp-503) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/DecimalBufferedSumAggregator.java:65]: Jun, can you research this one offline?  No need to hold the submit, but would be nice to have an explanation for the 64.

Comment (dgomezferro) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/DecimalBufferedSumAggregator.java:65]: Please make this part of the serialized message

**Review by dgomezferro — CHANGES_REQUESTED** [review]

**Review by dgomezferro — APPROVED** [review]

**Pull request merged**